bing_image_downloader/bing.py

```
@@ -31,73 +31,82 @@ def resize(url,size: tuple):
         if not self.resize:
 
             request = urllib.request.Request(link, None, self.headers)
             image = urllib.request.urlopen(request, timeout=self.timeout).read()
             if not imghdr.what(None, image):
-                print('[Error]Invalid image, not saving {}\n'.format(link))
+                self.log('[Error]Invalid image, not saving {}\n'.format(link))
                 raise ValueError('Invalid image, not saving {}\n'.format(link))
             with open(str(file_path), 'wb') as f:
                 f.write(image)
         elif self.resize:
             request = urllib.request.Request(link, None, self.headers)
```

Ruff (0.14.10)

Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)